genie-agent/src/main/java/com/netflix/genie/agent/execution/ExecutionContext.java

```diff
@@ -186,4 +187,33 @@ void addStateActionError(
      */
     @Nullable
     String getClaimedJobId();
+
+    /**
+     * If the job was killed, return the provenance of the kill request.
+     *
+     * @return null if the job was not killed, or else the provenance of the kill signal
+     */
+    @Nullable
+    KillProvenance getJobKillProvenance();
```

Comment:
Would `KillSource` be simpler? Less to spell all over the place

Reply:
I guess, but provenance sounds so much more pompous!